core/src/main/java/io/questdb/cairo/vm/MemoryPMARImpl.java

```diff
@@ -138,8 +139,8 @@ public void switchTo(FilesFacade ff, long fd, long extendSegmentSize, long offse
     }
 
     public void sync(boolean async) {
-        if (pageAddress != 0 && commitMode != CommitMode.NOSYNC) {
-            ff.msync(pageAddress, getPageSize(), commitMode == CommitMode.ASYNC);
+        if (pageAddress != 0) {
```

Comment:
is this change intentional? this will cause `msync()` even when `CairoConfig` says NOSYNC

Reply:
yeah, it is intentional. The logic is when signature is `sync(boolean)` we are committing, the question is not "if" it is "how". This function is called from a site, that has a check if NOSYNC is set or not.